panacea/query_client.go

```diff
@@ -0,0 +1,105 @@
+package panacea
+
+import (
+	"context"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/medibloc/panacea-core/v2/types/compkey"
+	aoltypes "github.com/medibloc/panacea-core/v2/x/aol/types"
+)
+
+type QueryClient struct {
+	rpcClient         *RpcClient
+	interfaceRegistry codectypes.InterfaceRegistry
+}
+
+// NewQueryClient set QueryClient with rpcClient & and returns, if successful,
+// a QueryClient that can be used to add query function.
+func NewQueryClient(ctx context.Context, chainID, rpcAddr string, trustedBlockHeight int, trustedBlockHash []byte) (*QueryClient, error) {
+	rpcClient, err := NewRpcClient(ctx, chainID, rpcAddr, trustedBlockHeight, trustedBlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return &QueryClient{
+		rpcClient:         rpcClient,
+		interfaceRegistry: makeInterfaceRegistry(),
+	}, nil
+}
+
+// Below are examples of query function that use GetStoreData function to verify queried result.
+// Need to set storeKey and key inside the query function, and change type to expected type.
+
+// GetAccount returns account from address.
+func (c QueryClient) GetAccount(address string) (authtypes.AccountI, error) {
+	acc, err := sdk.GetFromBech32(address, "panacea")
+	if err != nil {
+		return nil, err
+	}
+
+	key := authtypes.AddressStoreKey(acc)
+	bz, err := c.rpcClient.GetStoreData(context.Background(), authtypes.StoreKey, key)
+	if err != nil {
+		return nil, err
+	}
+
+	var accountAny codectypes.Any
+	err = accountAny.Unmarshal(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	var account authtypes.AccountI
+	err = c.interfaceRegistry.UnpackAny(&accountAny, &account)
+	if err != nil {
+		return nil, err
+	}
+
+	return account, nil
+}
+
+// GetBalance returns balance from address.
+func (c QueryClient) GetBalance(address string) (sdk.Coin, error) {
+	acc, err := sdk.GetFromBech32(address, "panacea")
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	var denom = "umed"
+	key := append(banktypes.BalancesPrefix, append(acc, []byte(denom)...)...)
+
+	bz, err := c.rpcClient.GetStoreData(context.Background(), banktypes.StoreKey, key)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	var balance sdk.Coin
+	err = balance.Unmarshal(bz)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+
+	return balance, nil
+}
+
+// GetTopic returns topic from address and topicName.
+func (c QueryClient) GetTopic(address string, topicName string) (aoltypes.Topic, error) {
+	acc, err := sdk.GetFromBech32(address, "panacea")
```

Comment:
Let's merge these panacea-dependent things into panacea package later.

Reply:
I don't understand exactly, so can you explain in detail?

Reply:
What I meant is that the logic related to panacea should be put in the panacea package.
Like this. (https://github.com/medibloc/panacea-coracle/tree/main/panacea)

Reply:
Oh by the way, it's already in the panacea package. :)

Reply:
So, what if you create an `account.go` file and make it implements like this:
Because I think It will be able to use these address converters often.

account.go
```go
const prefix = "panacea"

func GetAccAddressFromBech32(address string) (addr sdk.AccAddress, err error) {
  return sdk.GetFromBech32(address, prefix)
}
```

Reply:
Thanks, I agree. I reflect in the commit [60b9447](https://github.com/medibloc/panacea-doracle/pull/10/commits/60b94479345c6386c8704ce1d15fe3612605900e).

Reply:
The commit you shared does not reflect any modifications to the content.

Reply:
I found this commit. :)
https://github.com/medibloc/panacea-doracle/pull/10/commits/75b1490b373ee04860d6a91cb5288f47b51a6400